### PR TITLE
Classic Theme Helper: Add Restaurant Menus Custom Post Type to Classic Theme Helper Package

### PR DESCRIPTION
**Pull request by coder-karen**
Fixes https://github.com/Automattic/vulcan/issues/633

## Proposed changes:

* Moved the `nova.php` content across to the Classic Theme Helper package, as well as associated Nova CSS and JS files.
* Changed how assets were enqueued in the main Nova_Restaurant class, to use the Assets class - consistent with other files in the Classic Theme Helper package.
* Resolved some Phan-flagged errors, suppressed others where it made sense to.
* I have tested requiring the package to make sure that where possible no additional changes should be needed in the package before requiring, in the next PR.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/633
pfwV0U-ke-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* The package isn't being used yet, but as the build process is changing the location of the JS files in particular, a code review to check the enqueued JS file paths would make sense.
* If you do generally want to test the feature to make sure nothing is being added via the package, you would need to install and activate a theme that supports Restaurant Menus such as Confit.
  *  Add several new menu items from the new Food Menu in the admin menu.
  * Add some new sections, and add menu items to specific sections (see the Menu Sections area in the menu editor). Ensure some sections are parents to other sections
  * Attempt adding multiple menu items at once using the 'Add Many Items' menu option.
  * Using quick edit from the main menu list for specific item, change the sections they belong to in some cases.
  * Try dragging and dropping menu items to different positions and sections.
  * Click save new order, everything should save correctly.
* If testing, make sure you test with the changes applied for both the Jetpack Mu WPcom and Jetpack (so on Simple using both generated commands below) for WoA and Simple sites, and just Jetpack for self-hosted.

### CODE REVIEW

File: projects/packages/classic-theme-helper/src/custom-post-types/class-nova-restaurant.php [hunk 1]
```diff
@@ -0,0 +1,1758 @@
...
+			// phpcs:enable WordPress.WP.GlobalVariablesOverride.Prohibited
+
+			$this->setup_menu_item_columns();
+
+			Assets::register_script(
+				'nova-menu-checkboxes',
+				'../../dist/custom-post-types/js/menu-checkboxes.js',
+				__FILE__,
+				array(
+					'dependencies' => array(
```
Comment (darssen) [inline projects/packages/classic-theme-helper/src/custom-post-types/class-nova-restaurant.php]: Is jQuery needed here? I don't see it nova.php in trunk https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/custom-post-types/nova.php#L587

File: projects/plugins/jetpack/modules/custom-post-types/nova.php [hunk 2]
```diff
@@ -28,1717 +28,1721 @@
...
-	 * @param array $menu_item_loop_markup Array of markup for the menu items.
-	 *
-	 * @return self
-	 */
-	public static function init( $menu_item_loop_markup = array() ) {
-		static $instance = false;
+		/**
+		 * Current term ID of a loop of menu items.
+		 *
+		 * @var bool|int
```
Comment (darssen) [inline projects/plugins/jetpack/modules/custom-post-types/nova.php:81]: `menu_item_loop_current_term` seems to also be able to be of type `Wp_Term` after the assignment [here](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/custom-post-types/nova.php#L1588).

Also there are usages later that could result in Warnings (see 7ffe4564fc4c2a609c7f3ecbb525f372-logstash)

**Review by darssen — COMMENTED** [review]
Hi Karen, many thanks for these changes.
After a code review, the changes look mostly good. 

I left a couple of inline comments. One related to enqueuing and the other to existing unsafe usage of `menu_item_loop_current_term`. I thought we could use this PR to fix them, but not really necessary.

File: projects/packages/classic-theme-helper/src/custom-post-types/class-nova-restaurant.php (hunk 1 above)
Comment (coder-karen) [reply projects/packages/classic-theme-helper/src/custom-post-types/class-nova-restaurant.php]: Good catch, I just missed removing it here. Fixed in bd443351cfe77a3b2186e44c8f6139c64157ca21

File: projects/plugins/jetpack/modules/custom-post-types/nova.php (hunk 2 above)
Comment (coder-karen) [reply projects/plugins/jetpack/modules/custom-post-types/nova.php:81]: Good catch. I've updated them in the package version of the class here: dae867c35a60ccd84e2af851ccfdd3280953c3ea, makes sense to update them now before requiring the package file, if possible. It doesn't seem to make sense (to me) to fix it in `nova.php` though, given that we'll be deprecating that in a couple of weeks (plus I don't want to introduce possibly unexpected side effects just before everyone leaves for holidays). What do you think?

Comment (coder-karen) [reply projects/plugins/jetpack/modules/custom-post-types/nova.php:81]: I am slightly contradicting myself though, as that's why I made the JS changes in the module files (to make sure issues could be caught and then updated in the package versions later), but I think I'm just trying to limit possible issues over the holidays.

**Review by darssen — APPROVED** [review]
Thanks for tackling the comments above. Changes look good.

Tests to ensure existing functionality didn't bring up any issues in the three different envs that I tested with. :ship:

**Pull request merged**